src/parser/mod.rs

```diff
@@ -7105,6 +7105,15 @@ impl<'a> Parser<'a> {
             // Hive lets you put table here regardless
             let table = self.parse_keyword(Keyword::TABLE);
             let table_name = self.parse_object_name()?;
+
+            let is_postgresql = dialect_of!(self is PostgreSqlDialect);
```

Comment:
According [to the docs](https://github.com/cipherstash/sqlparser-rs/tree/main#new-syntax), dialect specific syntax needs to be handled for both the dialect in question and the `GenericDialect`.

Reply:
I think that only applies when you need to make changes to the dialect itself. I don't think that matters here, but I'll see what the maintainer says when I raise it upstream.

I still need to check other dialects as well to verify that this needs to be dialect specific.

Reply:
Isn't this doing exactly that? Changing the dialect?